migrate/20210623152412_create_tables.go

```diff
@@ -0,0 +1,366 @@
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_bitcoin_exchanges (
+				transaction_hash    BYTEA PRIMARY KEY,
+				bitcoin_block_hash  BYTEA NOT NULL,
+				bitcoin_merkle_root BYTEA NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_private_messages (
+				transaction_hash     BYTEA PRIMARY KEY,
+				recipient_public_key BYTEA  NOT NULL,
+				encrypted_text       BYTEA  NOT NULL,
+				timestamp_nanos      BIGINT NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_submit_posts (
+				transaction_hash    BYTEA PRIMARY KEY,
+				post_hash_to_modify BYTEA  NOT NULL,
+				parent_stake_id     BYTEA  NOT NULL,
+				body                BYTEA  NOT NULL,
+				timestamp_nanos     BIGINT NOT NULL,
+				is_hidden           BOOL   NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_update_exchange_rates (
+				transaction_hash      BYTEA PRIMARY KEY,
+				usd_cents_per_bitcoin BIGINT NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_update_profiles (
+				transaction_hash         BYTEA PRIMARY KEY,
+				profile_public_key       BYTEA,
+				new_username             BYTEA,
+				new_description          BYTEA,
+				new_profile_pic          BYTEA,
+				new_creator_basis_points BIGINT NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_follows (
+				transaction_hash    BYTEA PRIMARY KEY,
+				followed_public_key BYTEA NOT NULL,
+				is_unfollow         BOOL NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_likes (
+				transaction_hash BYTEA PRIMARY KEY,
+				liked_post_hash  BYTEA NOT NULL,
+				is_unlike        BOOL NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_creator_coins (
+				transaction_hash                BYTEA PRIMARY KEY,
+				profile_public_key              BYTEA NOT NULL,
+				operation_type                  SMALLINT NOT NULL,
+				bit_clout_to_sell_nanos         BIGINT NOT NULL,
+				creator_coin_to_sell_nanos      BIGINT NOT NULL,
+				bit_clout_to_add_nanos          BIGINT NOT NULL,
+				min_bit_clout_expected_nanos    BIGINT NOT NULL,
+				min_creator_coin_expected_nanos BIGINT NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_creator_coin_transfers (
+				transaction_hash               BYTEA PRIMARY KEY,
+				profile_public_key             BYTEA NOT NULL,
+				creator_coin_to_transfer_nanos BIGINT NOT NULL,
+				receiver_public_key            BYTEA NOT NULL
+			);
+		`)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Exec(`
+			CREATE TABLE pg_metadata_swap_identities (
```

Comment:
How does a metadata link back to its txn if it's in another table? Is there a foreign key constraint or something?

Reply:
Nevermind dumb question.